numpy/polynomial/tests/test_chebyshev.py

```diff
@@ -111,6 +112,21 @@ def test_chebdiv(self):
                 res = cheb.chebadd(cheb.chebmul(quo, ci), rem)
                 assert_equal(trim(res), trim(tgt), err_msg=msg)
 
+    def test_chebpow(self):
+        for i in range(5):
+            for j in range(5):
+                msg = "At i=%d, j=%d" % (i, j)
+                c = list(range(i+1))
+                [c] = pu.as_series([c])
+                power = j+1
+                zs = cheb._cseries_to_zseries(c)
+                prd = zs
+                for i in range(2, power + 1):
+                    prd = np.convolve(prd, zs)
```

Comment:
Hi @eric-wieser, thanks for the review. I worked my way down alphabetically, so I began with this function, but Hermite_E did not have the `_cseries_to_zseries` method, making reuse of this test unfeasible. So, I decided it would probably be best to stick as closely as possible to how each `pow` function performed its calculation.

Reply:
Ideally your tests shouldn't match your implementation - writing the same function in two places doesn't really test anything - and you probably shouldn't be calling private (underscore-led) functions in these tests.

As someone reading the tests, I can look at `test_hermpow` and think _"yes, I expect power to be repeated multiplication"_. I need to know about internal implementation details to understand what this test does.

Reply:
Makes sense to me, and I should've realize using private methods doesn't really make sense. I'll give this a little ponder and see if I can't come up with something more appropriate.